plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -1421,6 +1421,7 @@ static TupleDomain<ColumnDescriptor> getParquetTupleDomain(Map<List<String>, Col
         }
 
         Map<Integer, ColumnDescriptor> descriptorsById = descriptorsByPath.values().stream()
+                .filter(descriptor -> descriptor.getPrimitiveType().getId() != null)
```

Comment:
Will this skip predicate pushdown when field ids are not present in parquet footer schema ?

Reply:
I guess yes, because we can't push predicate to parquet when we don't know field ids

Reply:
I think so. We have the following logic:
https://github.com/trinodb/trino/blob/623bcc2fb328665467d12ae02cb8700d6ae0a389/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java#L1428-L1434

Reply:
I saw that we're able to generate ids from `nameMapping` for migrated tables from hive where the parquet files originally had no field ids. I'm wondering what's different in the pyarrow case which causes `ParquetSchemaUtil.applyNameMapping` to not provide ids. Is it just because of the contents of `schema.name-mapping.default` and not because of anything special about the parquet file from pyarrow ?

Reply:
Right. So, we can reproduce the issue with Hive connector if we directly add the file to Iceberg table without `migrate` procedure.

Reply:
ya, i think it'll be better to keep the test based on hive connector produced files. We'll be able to avoid the pyarrow specific test resources and someone looking into this scenario in future will not need to know anything outside Trino.

Reply:
Updated the test to create a Parquet file with Hive connector on the fly.